src/level_setup.m

```diff
@@ -0,0 +1,94 @@
+function level_collection = level_setup(directory)
+% build_level reads a directory containing level files. Level files are 
+% .txt files containing the objects to be placed in the
+% level. 
+
+% Input level_name = the name of the .txt file
+% Output level_collection = a Level array containing all levels in
+% directory.
+
+addpath(directory);
+D = dir(directory);
+ii=1;
+for nn = 1:length(D)
+    if D(nn).isdir
+        continue
+    else
+        [C,level_name] = read_level(D(nn));
+        level_collection(ii) = build_level(C,level_name);
+        ii = ii+1;
+    end
+    
+end
+
+end
+
+function [C,level_name] = read_level(D)
+level_name = D.name;
+fid = fopen(level_name);
+C = textscan(fid,'%s %f %f %f %f','headerLines',1,'TreatAsEmpty','none');
+fclose(fid);
+end
+
+function level_collection = build_level(C,level_name)
+lvl = Level();
+lvl.set_id(str2num(level_name(6)));
+j=1;
+for qq = 1:size(C{1},1)
+    switch C{1}{qq}
+        case 'rm'
+            object{j} = MirrorFactory.build_rectangle(C{2}(qq),C{3}(qq),[C{4}(qq),C{5}(qq)]);
```

Comment:
Try to create a function that handles constructing components from the data structure and adding them to the level list, so you don't repeat too much logic here